### ISSUE TITLE
Setup TravisCI to successfully build using firebase-config.js

### ISSUE THREAD

**Issue by MikeTheCanuck** [MikeTheCanuck/WhoDidITalkTo#60]
TravisCI will of course require the firebase config including the "secrets" - until then, all [builds will fail](https://travis-ci.org/MikeTheCanuck/WhoDidITalkTo/builds/304559444):

```
5.50s$ npm run build
> timeline@0.1.0 build /home/travis/build/MikeTheCanuck/WhoDidITalkTo
> react-scripts build
Creating an optimized production build...
Failed to compile.
Module not found: Error: Can't resolve './firebase-config' in '/home/travis/build/MikeTheCanuck/WhoDidITalkTo/src'
```

**Comment by MikeTheCanuck** [comment]
There are a few options:
1. Publish the current `firebase-config.js` to the repo.
2. Setup env vars in TravisCI and create a `firebase-config.js` that relies on env vars for those secrets.
3. Configure `.travis.yml` to run a script that downloads a fully-configured `firebase-config.js` from firebase Storage.

While I've been attempting to minimize the footprint of my app's secrets across the interwebs, it finally dawned on me last night to check the build, and it turns out that the deployed `main.******.js` which downloads to the client's browser includes all the secrets (as I kinda knew, but was politely ignoring).  Impossible for a client-only application *not* to include the symmetric secrets, I know, but I was really hoping Google had come up with some new approach that wasn't this dramatically insecure.

Further, even though I've enabled authentication around the data-access portions of the application, that still means the bare application (including secrets) are sent to anyone who requests the URI for the app.

So any option I choose will really just be testing the trade-off between not *also* publishing the secrets directly in the github repo and how much more complicated I want to make it for anyone who wishes to deploy this app for themselves (on their own Firebase project).

## Option 1: push firebase-config.js to the github repo
- **Pro:** Travis gets access immediately and directly to the config and secrets
- **Con:** All secrets are trivially available to the world in the repo
- **Con:** Others who wish to deploy their own Firebase project will have to edit the config file in detail

## Option 2: populate an env var-laden firebase-config.js in the repo
- **Pro:** the config file becomes reusable by myself (in dev), Travis (in build/test/deploy) and other consumers (in their own Firebase project)
- **Con:** Setting up the environment variables (e.g. in an `env.sh` script that must be `source`d every time you build) is just as much trouble for newbies as overwriting hard-coded strings in a config file, and non-Travis-based deploys (e.g. using `firebase deploy`) might be harder for all concerned

## Option 3: download firebase-config.js from firebase Storage
- **Pro:** preserves the current status quo, 
- **Con:** requires that the config file be published anonymously, or that Travis somehow acquire credentials to authenticate to firebase Storage
- **Con:** even if I figure out how to get Travis access to the file, that means that I'll have to publish the Storage location in the `.travis.yml` file for everyone to find in my repo
- **Con:** unless I can figure a way to prevent new entrants from auto-enrolling themselves in the firebase project's authentication, they will be able to download even the authenticated config file

## Option 4: write an init script that overwrites the `firebase-config.js` file for newbies
- **Pro:** I can publish the hardcoded configuration directly in the github repo, and any user who is following my README instructions will overwrite my configuration values with their own values
- **Con:** Users could accidentally attempt to use my configuration values, and unless I can prevent them from auto-enrolling in my project, they'll be able to see my private data (and potentially add their own 'garbage' data).

**Comment by MikeTheCanuck** [comment]
It appears that, unless I can become comfortable with making it possible for anyone to view my private notes in my data, I can't think of a practical solution that both reasonably prevents data read access *and* doesn't create inordinate hoops for others to jump through if they wish to deploy this code to their own firebase project.

**Comment by MikeTheCanuck** [comment]
With #62 addressed, I should no longer have to worry about publishing the firebase config - at least as far as inadvertently exposing my own private data to others.  Thus, I can now remove `src/firebase-config.js` from `.gitignore`.

The only real side-effect I have to deal with at this point is revising the `README.md` instructions on how to fork/clone this repo and configure a deployment for your own usage.  I know I started on those instructions somewhere...they must be in a branch (that I hope I haven't deleted)...

Well, you know what - let's worry about the fork/clone/deploy documentation in a separate issue #67.

**Comment by MikeTheCanuck** [comment]
Publishing the `src/firebase-config.js` in the repo meant we got a successful build!

**Comment by MikeTheCanuck** [comment]
Added `sudo: false` directive to `.travis.yml` reduced the build time from ~2m45s to ~1m14s.

**Comment by MikeTheCanuck** [comment]
Swapped `yarn build` for `npm run build` without issue, and swapped `yarn test` for `npm run test:cover:travis` without issue as well.

**Comment by MikeTheCanuck** [comment]
Reconfigured `node_js` to use `lts/*` - which currently resolves to:

```
$ nvm install lts/*
Downloading and installing node v8.9.1...
```

The build and test continue to succeed.

**Comment by MikeTheCanuck** [comment]
Added `deploy` to `.travis.yml` earned a message "Skipping a deployment with the firebase provider because the current build is a pull request".

This is as close as we'll be able to get with this TravisCI configuration without pulling the PR to `master`, so cross fingers and let's hope this doesn't get too messy!

**Comment by MikeTheCanuck** [comment]
This work to setup TravisCI build, test and deploy was enabled heavily by [this article](https://codeburst.io/learning-travis-ci-with-firebase-react-part-2-28a131913e28), though thankfully TravisCI has progressed sufficiently since its publication that yarn no longer needs to be explicitly installed by a Travis supplicant.

**Comment by MikeTheCanuck** [comment]
Solved by #68 .
